sourcePage.js

```diff
@@ -17,61 +22,91 @@ function getPage() {
+        // Finds classes based on major
+        switch(selectedMajor) {
+            case 'computer-science':
+                concentration = computerScience(className);
                 break;
         }
 
-        html += (element + ' ' + concentration + '\n');
+        html += (className + ' (' + concentration + ')\n');
     }
     return html;
 }
 
+function computerScience(className) {
+    var concentration; // Local variable to hold concentration
+
+    switch(className) {
+        case 'MATH 1200': case 'MATH 1201': case 'MATH 1300': case 'MATH 1301': case 'MATH 2300': case 'MATH 2410': case 'MATH 2600': case 'MATH 2500': case 'MATH 2501':
+            concentration = "Math: Calculus/Linear Algebra";
+            break;
+        case 'MATH 2810': case 'MATH 2820': case 'MATH 3640':
+            concentration = "Math: Statistics";
+            break;
+        case 'MATH 2420':
+            concentration = "Math: Elective Course";
+            break;
+        case 'BSCI 1101': case 'BSCI 1101L': case 'BSCI 1510': case 'BSCI 1510L': case 'BSCI 1511': case 'BSCI 1511L': case 'BSCI 2218': case 'BSCI 2219': case 'CHEM 1601': case 'CHEM 1601L': case 'CHEM 1602': case 'CHEM 1602L': case 'MSE 1500': case 'MSE 1500L': case 'PHYS 1601': case 'PHYS 1601L': case 'PHYS 1602': case 'PHYS 1602L':
+            concentration = "Science";
+            break;
+        case 'ES 1401': case 'ES 1402': case 'ES 1403':
+            concentration = "Introduction to Engineering";
+            break;
+        case 'CS 1101': case 'CS 1104': case 'CS 2201': case 'CS 3251': case 'CS 3270': 
+            concentration = "CS Core: Software/Problem Solving";
+            break;
+        case 'EECE 2123': case 'EECE 2123L': case 'CS 3281:': 
+            concentration = "CS Core: Hardware/Systems";
+            break;
+        case 'CS 2212': case 'CS 3250':
+            concentration = "CS Core: Foundations";
+            break;
+        case 'CS 4260': case 'CS 4278': case 'CS 3860': case 'CS 3861':
+            concentration = "CS Depth";
+            break;
+        case 'EECE 4353': case 'EECE 4354': case 'EECE 4376':
+            concentration = "CS Depth, Technical Electives";
+            break;
+        case 'MATH 3320': case 'MATH 3620': case 'MATH 4600': case 'MATH 4620':
+            concentration = "Math: Elective, CS Depth, Technical Electives";
+            break;
+        case 'CS 3259': case 'CS 3892': case 'CS 4269': case 'CS 4279': case 'CS 4287':
+            concentration = "CS Project";
+            break;
+        case 'CS 4959':
+            concentration = "Computer Science Seminar";
+            break;
+        case 'CS 1151:':
+            concentration = "Liberal Arts Core or Open Elective";
+            break;
+        case 'MATH 2610': case 'MATH 2821': 
+            concentration = "Math: Elective Course"
+        default:
+            concentration = "Open Elective";
+            break;
+    }
```

Comment:
Should we do this for every major? Handle every class as if it can fulfill multiple things?

Reply:
I think that's a good idea - I think it would be best to include multiple requirements that a class can fulfill

Reply:
I did this for CS and EE, it's a pain but maybe we can finish all engineering by school?